src/Functions/URL/cutURLParameter.cpp

```diff
@@ -1,82 +1,183 @@
+#include <Columns/ColumnArray.h>
+#include <Columns/ColumnConst.h>
+#include <Columns/ColumnString.h>
+#include <Columns/ColumnFixedString.h>
+#include <Columns/ColumnVector.h>
+#include <DataTypes/DataTypeArray.h>
+#include <DataTypes/DataTypeString.h>
+#include <DataTypes/DataTypesNumber.h>
+#include <Functions/FunctionHelpers.h>
+#include <Functions/IFunction.h>
+#include <IO/WriteHelpers.h>
+#include <Interpreters/Context.h>
+#include <base/StringRef.h>
 #include <Functions/FunctionFactory.h>
-#include <Functions/FunctionsStringSearchToString.h>
 #include <base/find_symbols.h>
 
 namespace DB
 {
 
-struct CutURLParameterImpl
+namespace ErrorCodes
 {
+    extern const int ILLEGAL_TYPE_OF_ARGUMENT;
+    extern const int ILLEGAL_COLUMN;
+}
+
+class FunctionCutURLParameter : public IFunction
```

Comment:
I was first quite reluctant about copypasting `FunctionsStringSearchToString`. I thought, perhaps we can provide Array arguments also to all other users of `FunctionsStringSearchToString`, i.e. extend base class `FunctionsStringSearchToString` with the below logic.

Turns out that 1. all other "URL" functions (https://clickhouse.com/docs/en/sql-reference/functions/url-functions/) don't inherit from `FunctionsStringSearchToString` and 2. the SQL functions which do inherit from `FunctionsStringSearchToString` are in scattered categories and in general don't benefit from Array arguments.

So it is fine to copy-paste `FunctionsStringSearchToString`.